modules/fluid_properties/include/fluidproperties/SinglePhaseFluidProperties.h

```diff
@@ -39,7 +39,14 @@
                                              DualReal & d##want##d1,                               \
                                              DualReal & d##want##d2) const                         \
   {                                                                                                \
-    fluidPropError(name(), ": ", __PRETTY_FUNCTION__, " derivative derivatives not implemented."); \
+    fluidPropError("The fluid properties class '",                                                 \
+                   type(),                                                                         \
+                   "' has not implemented the derivatives for the automatic "                      \
+                   "differentiation-enabled method below. You can choose to neglect the "          \
+                   "derivatives by passing the 'allow_incorrect_jacobian' parameter. If your "     \
+                   "application requires this method, you "                                        \
+                   "must either implement it or use a different fluid properties class.\n\n",      \
```

Comment:
```suggestion
                   "' has not implemented the automatic "                      \
                   "differentiation-enabled method below. You can choose to neglect the "          \
                   "derivatives by setting the 'allow_imperfect_jacobians' parameter. If your "     \
                   "application requires this method, you "                                        \
                   "must either implement it or use a different fluid properties class.\n\n",      \
```

Reply:
However, I think a better design would be to move that bit about setting that parameter into `fluidPropError`, but only into the error message, not the warning, since for the latter, they've already set that parameter. Also bonus points (and I can do this as follow-up): `fluidPropError` is a vague name that says nothing that's it about the derivatives not being implemented, so I suggest we replace `fluidPropError(args)` with `unimplementedDerivativeMethod(pretty_name)` and `unimplementedNonderivativeMethod(pretty_name)`. Also I'd also add in the warning a note about the derivatives being approximate now.

Reply:
ok I did that. You're not wrong for the clarity. I think the idea was to use the same method / warning everywhere for these errors but it only ever got used for derivatives (in the source of the module at least. Maybe I m going to hit relap7 errors next. We'll see)